src/server/logs/logs.go

```diff
@@ -103,12 +106,59 @@ func (ls LogService) GetLogs(ctx context.Context, request *logs.GetLogsRequest,
 						Log: &logs.LogMessage{
 							LogType: &logs.LogMessage_Verbatim{
 								Verbatim: &logs.VerbatimLogMessage{
-									Line: []byte(e.Line),
+									Line:      []byte(e.Line),
+									Timestamp: timestamppb.New(e.Timestamp),
 								},
 							},
 						},
 					},
 				}
+			case logs.LogFormat_LOG_FORMAT_PARSED_JSON:
+				resp = &logs.GetLogsResponse{
+					ResponseType: &logs.GetLogsResponse_Log{
+						Log: &logs.LogMessage{
+							LogType: &logs.LogMessage_Json{
+								Json: &logs.ParsedJSONLogMessage{
+									Verbatim: &logs.VerbatimLogMessage{
+										Line:      []byte(e.Line),
+										Timestamp: timestamppb.New(e.Timestamp),
+									},
+									NativeTimestamp: timestamppb.New(e.Timestamp),
+								},
+							},
+						},
+					},
+				}
+				jsonStruct := new(structpb.Struct)
+				if err := jsonStruct.UnmarshalJSON([]byte(e.Line)); err == nil {
+					resp.GetLog().GetJson().Object = jsonStruct
```

Comment:
The typical pattern one sees is to bail `if err != nil`. Do you want to continue in case of an error? If you do, do you want to log the error at least? I’m not certain if an empty response is terribly useful.

Reply:
https://go.dev/wiki/CodeReviewComments#indent-error-flow